unlock-app/src/middlewares/currencyConversionMiddleware.js

```diff
@@ -1,33 +1,30 @@
 // TODO convert to fetch instead of using axios when it is supported well enough
 /* eslint promise/prefer-await-to-then: 0 */
 
-import axios from 'axios'
 import { setConversionRate } from '../actions/currencyConvert'
 import configure from '../config'
+import CurrencyLookupService from '../services/currencyLookupService'
 
 const { services } = configure(global)
 const currencyPriceLookupURI = services.currencyPriceLookup
+const retryInterval = 10000
 
 export default store => {
-  axios
-    .get(currencyPriceLookupURI)
-    .then(info =>
-      store.dispatch(
-        setConversionRate(info.data.data.currency, info.data.data.amount)
-      )
-    )
+  const currencyLookupService = new CurrencyLookupService(
+    currencyPriceLookupURI
+  )
+
+  currencyLookupService
+    .lookupPrice('ETH', 'USD')
+    .then(info => store.dispatch(setConversionRate(info.currency, info.amount)))
+
   setInterval(() => {
-    axios.get(currencyPriceLookupURI).then(info => {
-      const {
-        data: {
-          data: { currency, amount },
-        },
-      } = info
-      const current = store.getState().currency[currency]
-      if (current === +amount) return
+    currencyLookupService.lookupPrice('ETH', 'USD').then(info => {
```

Comment:
It looks like this call is duplicate with the above? Any way to refactor to avoid that?

Reply:
I'd like to hold off on this change at the moment, to see if a larger case for reuse emerges

Reply:
works for me